runtime_api.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+// addRuntimeAPI exposes additional functions in the runtime
+// package that may be helpful when hiding information
+// during execution is required.
+func addRuntimeAPI(filename string, file *ast.File) {
+	switchPanicPrints := func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if !ok {
+			return true
+		}
+		id, ok := call.Fun.(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		if id.Name == "print" {
+			id.Name = "panicprint"
+			return false
+		}
+
+		return true
+	}
+
+	switch filename {
+	case "debug.go":
+		// Add HidePanics function and internal hidePanics variable
+		file.Decls = append(file.Decls, hidePanicsDecls...)
+	case "error.go":
+		// Add a function panicprint, that does nothing if panics are
+		// hidden, otherwise forwards arguments to printany to print them
+		// as normal.
+		//
+		// We will also add two statements to printany:
+		// 1. An if statement that returns early if panics are hidden, as
+		//    printany is called directly when printing panics
+		// 2. An additional case statement that handles printing runtime.hex
+		//    values. Without this case statement, the default case will print
+		//    the runtime.hex values in a way not consistent with normal panic
+		//    outputs
```

Comment:
I don't understand this second case - what is a runtime.hex? is this tested in your testscript?

Reply:
`runtime.hex` is a type that just tells the runtime to print a `uint64` as hex: https://github.com/golang/go/blob/master/src/runtime/print.go#L15

It is not handled by specifically by `printany`, meaning when it is passed into `printany` it is printed in a weird format. Here is what a panic looks like without this added case for `runtime.hex`:

```
panic: ya like jazz?
            
goroutine 1 [running]:
main.main()
    z0.go:11 +(runtime.hex) (0x463380,0x7fff65557470)
```

I tried to have `panicprint` just forward all calls to `print` to make it easier, but it turns out `print` prints interfaces as their address in memory, and `panicprint` takes in a variardic number of `interface{}`s. So using `printany` and adding a single case was easier and means less changes to the runtime.

Reply:
This case is not tested per say, not sure how I'd do that, but it makes the output of panics the same as panic output from an unmodified runtime. If you want cases added to the the panic test I can, just not sure what to add myself.